34.md

```diff
@@ -181,6 +182,49 @@ The Status event with the largest created_at date is valid.
 
 The Status of a patch-revision defaults to either that of the root-patch, or `1632` (Closed) if the root-patch's Status is `1631` and the patch-revision isn't tagged in the `1631` event.
 
+## Git client nostr plugin (git-remote-nostr)
+
+Definition: Clone URL - a git remote URL listed under the `clone` tag of the repositories kind `30617` event
+
+The offical git client supports interacting with 'remotes' using custom protocols via [git-remote-helpers](https://git-scm.com/docs/gitremote-helpers).
+
+An implementation of git-remote-nostr SHOULD:
+
+### 1. Use Git Remote Nostr URL Format
+
+Struture:
+
+- `nostr://` - git syntax for a protocol URL
+- `<user>@` - optional ssh user for interacting with Clone URLs
+- `<protocol>/` - optional prefered protocol for interacting with Clone URLs. eg. ssh or https
+- kind `30617` event coordinate in format:
+  - `<naddr>`
+  - `<npub|nip05-address>/<identifier>`
+  - `<npub|nip05-address>/<relay-hint*>/<identifier>`
+
+\*relay hints SHOULD be [URL encoded](https://www.w3.org/Addressing/URL/4_Recommentations.html) eg `ws%3A%2F%2Funencrypted.com` and may omit `wss://` for brevity and readability.
+
+for example `nostr://dan@gitworkshop.dev/ngit` or `nostr://npub15qydau2hjma6ngxkl2cyar74wzyjshvl65za5k5rl69264ar2exs5cyejr/relay.damus.io/ngit`
+
```

Comment:
I think it should be mandatory to use nip05 relays if somebody wants to use their nip05 in a git remote url. It should throw error if either:
- there are no nip05 relays and no relay hint in the url
- the repo event is not found on any of the nip05 or hint relays
But I have no strong opinions on this. It's possible to use the bootstrap/fallback relays to find the repo (30617) event.

Anyway it should be clear that the NIP-05 relays or the hint relays are used to find the repo event, which contains the repo relays. All subsequent operations go to the repo relays.

Reply:
good clients that show the nostr git url should include a relay hint inline if the announcement event isn't on, at least one of, the nip05 relays. this should be a client implementation detail.